app/scripts/hglib.js

```diff
@@ -91,14 +112,19 @@ const launch = (element, config, options) => {
  *
  * @return  {Object}  Newly created HiGlass component.
  */
-export const viewer = (element, viewConfig, options) => {
+export const viewer = (element, viewConfig, options,
+  refAsPromise = false) => {
   /**
    * Available options:
    *
    *  bounded: [true/false]
    *      Fit the container to the bounds of the element
    */
-  const hg = launch(element, viewConfig, options);
 
-  return hg.api;
+  if (refAsPromise) {
+    const p = launch(element, viewConfig, options, refAsPromise);
```

Comment:
```suggestion
    return launch(element, viewConfig, options, asPromise).then(hg => hg.api);
```

Let's same some of these invaluable bytes 🤓